### PR TITLE
Initial API

### PR DESCRIPTION
**Pull request by FreeMasen**
This will represent the initial release candidate aka 0.1.0

### CODE REVIEW

File: client.lua [hunk 1]
```diff
@@ -0,0 +1,20 @@
+local lunch = require 'luncheon'
```
Comment (cjswedes) [inline client.lua:1]: This file seems to be basically repeated in the `examples` directory. Do we need both?

File: luncheon/headers.lua [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    end
+end
+
+---Serialize a key value pair
+---@param key string
+---@param value string
+---@return string
+function Headers.serialize_header(key, value)
+    if type(value) == 'table' then
+        value = value[#value]
```
Comment (cjswedes) [inline luncheon/headers.lua:36]: If there is more than one value for a header, only the last one added will be serialized. I think all the values can be included as a comma separated list. I faced this small issue while working with lustre. I would be fine leaving as is for now, but at some point we probably want to serialize multiple values correctly.

**Review by cjswedes — COMMENTED** [review]
LGTM

File: client.lua (hunk 1 above)
Comment (FreeMasen) [reply client.lua:1]: Good call, it really should be part of the .gitignore

File: luncheon/headers.lua (hunk 2 above)
Comment (FreeMasen) [reply luncheon/headers.lua:36]: Good catch. I'll add an issue for this

**Pull request merged**